### PR TITLE
chore: yarn global cache

### PR DESCRIPTION
**Pull request by dbrugger**
Contributes to https://github.com/carbon-design-system/carbon/issues/14235

#### Changelog

**Changed**

- Use yarn global cache
- Use npm registry

**Removed**

- Unneeded repo local yarn cache

#### Testing / Reviewing

`yarn install`

Build time improvements:
- Checkout step in GitHub Action down from 26s to 5-7s.
- test:e2e down from 2m 25s to 2m 7s
- lint unchanged at 2m 25s
- avt down from 27m 34s to 21m 18s
- test up from 19m 27s to 20m 41s

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,5 +1,11 @@
+enableGlobalCache: true
```
Comment (tay1orjones) [inline .yarnrc.yml:1]: This defaults to `true`, so should already be on.

File: .yarnrc.yml [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+
+httpTimeout: 300000
```
Comment (tay1orjones) [inline .yarnrc.yml:3]: What's the benefit of increasing the timeout from the default (60000) to 300000?

File: .yarnrc.yml [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+
 nodeLinker: node-modules
 
+npmRegistryServer: 'https://registry.npmjs.org/'
```
Comment (tay1orjones) [inline .yarnrc.yml:7]: The default registry, `registry.yarnpkg.com` is just a CNAME of the npm registry. Is there any tangible benefit to not using the default?

https://yarnpkg.com/getting-started/qa#why-registryyarnpkgcom-does-facebook-track-us

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -14,40 +14,28 @@ concurrency:
...
-  format:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Use Node.js 18.x
-        uses: actions/setup-node@v3
-        with:
-          node-version: '18.x'
       - name: Install dependencies
-        run: yarn install --immutable --immutable-cache
```
Comment (tay1orjones) [inline .github/workflows/ci.yml:37]: We want CI jobs to fail if a change to the cache would be made as a result from this install. This helps ensure integrity of dependencies between local dev environments and CI.

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Could you clarify how these changes result in a smaller repo size or faster CI times? I have a few additional comments below, let me know what you think.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (dbrugger) [reply .github/workflows/ci.yml:37]: still does. yarn knows if it's running in a CI

File: .yarnrc.yml (hunk 3 above)
Comment (dbrugger) [reply .yarnrc.yml:7]: would prefer the official hostname for the registry, one less variable for security.

File: .yarnrc.yml (hunk 2 above)
Comment (dbrugger) [reply .yarnrc.yml:3]: not sure if it's needed, but saw that used in a couple of repos using yarn so might be some kind of best practice. fine to remove also.

File: .yarnrc.yml (hunk 1 above)
Comment (dbrugger) [reply .yarnrc.yml:1]: default unless some other settings are configured. this removes the local (in repo) cache that creates the tar balls in Git. global means the cache is in the home dir which is what we need to clean that up.

**Comment by tay1orjones** [comment]
Thanks again for taking some time to discuss this today! As we discussed I'll close this for now and we'll look at revisiting it after we land https://github.com/carbon-design-system/carbon/issues/14244.

**Pull request closed without merging**